app/models/enumeration/EventType.scala

```diff
@@ -83,7 +84,6 @@ object EventType extends Enumerable.Implicits {
       case evType1827 if api1827Events.contains(evType1827) => Some(ApiType.Api1827)
       case evType1829 if api1829Events.contains(evType1829) => Some(ApiType.Api1829)
       case evType1830 if api1830Events.contains(evType1830) => Some(ApiType.Api1830)
-      case _ => None
```

Comment:
If you remove this, why do we need following else cases ? 
`
  def saveUserAnswers(pstr: String, eventType: EventType, userAnswersJson: JsValue)(implicit ec: ExecutionContext): Future[Unit] = {
    EventType.postApiTypeByEventType(eventType) match {
      // TODO: Have discussion on potential for overwriting in Mongo.
      case Some(apiType) => eventReportCacheRepository.upsert(pstr, apiType, userAnswersJson)
      case _ => Future.failed(new NotFoundException(s"Not Found: ApiType not found for eventType ($eventType)"))
    }
  }
`

and 

`  def getUserAnswers(pstr: String, eventType: EventType)(implicit ec: ExecutionContext): Future[Option[JsObject]] = {
    EventType.postApiTypeByEventType(eventType) match {
      case Some(apiType) =>
        eventReportCacheRepository.getByKeys(Map("pstr" -> pstr, "apiTypes" -> apiType.toString))
          .map(_.map(_.as[JsObject]))
      case _ => Future.failed(new NotFoundException(s"Not Found: ApiType not found for eventType ($eventType)"))
    }
  }`

Reply:
Addressed here but with the unchecked annotation - not sure if we'd like to proceed with this approach: https://github.com/hmrc/pension-scheme-event-reporting/pull/36/commits/2c089af273be16623552a522c80200a00f9429a2